ThirdParty/ecec/src/params.c

```diff
@@ -327,76 +338,125 @@ ece_header_extract_params(const char* header) {
 }
 
 int
-ece_header_extract_aesgcm_crypto_params(const char* cryptoKeyHeader,
-                                        const char* encryptionHeader,
-                                        uint32_t* rs, ece_buf_t* salt,
-                                        ece_buf_t* rawSenderPubKey) {
-  int err = ECE_OK;
+ece_aes128gcm_payload_extract_params(const uint8_t* payload, size_t payloadLen,
+                                     const uint8_t** salt, size_t* saltLen,
+                                     const uint8_t** keyId, size_t* keyIdLen,
+                                     uint32_t* rs, const uint8_t** ciphertext,
+                                     size_t* ciphertextLen) {
+  if (payloadLen < ECE_AES128GCM_HEADER_LENGTH) {
+    return ECE_ERROR_SHORT_HEADER;
+  }
+
+  *salt = payload;
+  *saltLen = ECE_SALT_LENGTH;
+
+  *keyIdLen = payload[ECE_SALT_LENGTH + 4];
+  if (payloadLen < ECE_AES128GCM_HEADER_LENGTH + *keyIdLen) {
```

Comment:
Good catch, I think this should be `<=`. `payload` is the complete encrypted message, with the [header block](http://httpwg.org/http-extensions/draft-ietf-httpbis-encryption-encoding.html#rfc.section.2.1) and ciphertext. The ciphertext can't be empty, so we may as well fail here because `ece_webpush_aes128gcm_decrypt` will also fail.